frontend/src/components/search/SearchBar.tsx

```diff
@@ -1,179 +1,265 @@
+    useState<Array<RecentSearchProps>>();
+  const [facetValue, setFacetValue] = useState<FacetValueProps>({
+    display_name: "",
+    name: "",
+  });
+
+  const { data } = api.dataset.search.useQuery({
+    limit: query?.length > 1 ? 5 : 0,
+    query: query,
+    sort: "score desc, metadata_modified desc",
+    ...(facetValue.name
+      ? {
+          [facetName]:
+            facetName === "startYear" || facetName === "endYear"
+              ? facetValue.name
+              : [facetValue.name],
+        }
+      : {}),
 
+    facetsFields: `["regions", "sectors", "modes", "services", "indicator", "temporal_coverage_start", "temporal_coverage_end"]`,
+  });
+
+  const facets: any = {
+    regions: {
+      field: "in",
+      description: "a region, country or a city",
+      options: data?.facets?.regions?.items,
+      isMultiple: true,
+    },
+    startYear: {
+      field: "after",
+      description: "referencing data after an year",
+      options: (data?.facets?.temporal_coverage_start?.items as any[])
+        ?.map((d: any) => ({
+          ...d,
+          display_name: new Date(d.name)?.getFullYear(),
+          name: new Date(d.name)?.getFullYear(),
+        }))
+        .filter(
+          (obj, index, self) =>
+            index === self.findIndex((o) => o.name === obj.name)
+        ),
+    },
+    endYear: {
+      field: "before",
+      description: "referencing data before an year",
+      options: (data?.facets?.temporal_coverage_end?.items as any[])
+        ?.map((d: any) => ({
+          ...d,
+          display_name: new Date(d.name)?.getFullYear(),
+          name: new Date(d.name)?.getFullYear(),
+        }))
+        .filter(
+          (obj, index, self) =>
+            index === self.findIndex((o) => o.name === obj.name)
+        ),
+    },
+    sectors: {
+      field: "sector",
+      description: "road, rail, aviation, water transportation",
+      options: data?.facets?.sectors?.items,
+      isMultiple: true,
+    },
+    modes: {
+      field: "mode",
+      description: "car, 2W, 3W, multi-modal etc.",
+      options: data?.facets?.modes?.items,
+      isMultiple: true,
+    },
+    services: {
+      field: "service",
+      description: "passenger or freight",
+      options: data?.facets?.services?.items,
+      isMultiple: true,
+    },
+  };
+
+  /* handle click outside the search container to close command list */
   useEffect(() => {
     const handleClickOutside = (event: MouseEvent) => {
       if (
         commandRef.current &&
         !commandRef.current.contains(event.target as Node)
       ) {
-        if (!hideDatasetSuggestion) setShowCommandList(false);
+        setShowCommandList(false);
       }
     };
-
     document.addEventListener("mousedown", handleClickOutside);
     return () => {
       document.removeEventListener("mousedown", handleClickOutside);
     };
   }, []);
+  //get stored searches
+  useEffect(() => {
+    if (typeof window !== "undefined")
```

Comment:
You can remove it, the useEffect always is executed on the client side.

Reply:
At first, Window or localStorage is not available.:
![image](https://github.com/user-attachments/assets/47049f59-2954-43ca-935e-0637b3806d05)

Reply:
I said to you to remove the if not the useEffect.